src/lib/journey/callbacks/string-attribute/string-attribute-input.svelte

```diff
@@ -10,16 +10,17 @@
   import { interpolate } from '$lib/utilities/i18n.utilities';
 
   export let callback: AttributeInputCallback<string>;
+  export let firstInvalidInput: boolean;
   export let idx: number;
 
-  const inputName = callback?.payload?.input?.[0].name || `string-attr-${idx}`;
-  const isRequired = isInputRequired(callback);
-  const outputName = callback.getOutputByName('name', '');
-  const policies = callback.getPolicies();
-  const type = getInputTypeFromPolicies(policies);
-  const previousValue = callback?.getInputValue() as string;
-  const textInputLabel = callback.getPrompt();
-  const validationFailure = getAttributeValidationFailureText(callback);
+  let inputName = callback?.payload?.input?.[0].name || `string-attr-${idx}`;
+  let isRequired = isInputRequired(callback);
+  let outputName = callback.getOutputByName('name', '');
+  let policies = callback.getPolicies();
+  let type = getInputTypeFromPolicies(policies);
+  let previousValue = callback?.getInputValue() as string;
```

Comment:
I think the **previousValue** property is what we're using to render the existing value of the property returned by AM, where the is one. For example if we have a tree that is allowing the user to update their phone number, if they have provided it already and they already have a session (i.e. a "self service" tree). I think we should test with these kinds of trees as we move forward, to make this functionality easier to verify.

Reply:
Yeah, the register journey is a perfect example of this. If you try to register, but your submission has some errors in it, the `input` property will have the previous value in it.

Reply:
Ah ok so that's the same thing then, kind of. Understood